### DISCUSSION TITLE
Changing the simulation time step

### DISCUSSION THREAD

**Discussion by Chandan-Bharadwaj** [carla-simulator/carla#4378] [Q&A]
Hi everyone, 
I would like to know how to set the timesteps correctly in carla. we have a csv file running in 0.1 sec discrete timesteps and we would like to set the velocity of the vehicle as per the inputs from the csv file. 

We have tried changing the "settings.fixed_delta_seconds" as per the documentation: https://carla.readthedocs.io/en/latest/adv_synchrony_timestep/#simulation-time-step  
but there is no effect of that in our case. 
is there any other way forward ?

Also, please suggest us an easy way to verify that the time step is set correctly.

**Comment by corkyw10** [comment]
Can you share the full code you are using to set the timestep?

**Reply by Chandan-Bharadwaj** [reply]
To set the timesteps through FPS I have tried these two: 

![image](https://user-images.githubusercontent.com/85007096/125594334-4416abe3-2eb3-4a52-9ae7-9f2f9883166f.png)

![image](https://user-images.githubusercontent.com/85007096/125594527-2184d0a4-3c1f-48be-8d5e-bb74db15d6a6.png)

I am using the python API "manual_control.py" from examples, with a small addition of code which logs time onto a csv file ('log.csv'). This is the modified portion of the "manual_control.py":
```
def game_loop(args):
    pygame.init()
    pygame.font.init()
    world = None

    try:
        client = carla.Client(args.host, args.port)
        client.set_timeout(2.0)

        display = pygame.display.set_mode(
            (args.width, args.height),
            pygame.HWSURFACE | pygame.DOUBLEBUF)
        display.fill((0,0,0))
        pygame.display.flip()

        hud = HUD(args.width, args.height)
        world = World(client.get_world(), hud, args)
        controller = KeyboardControl(world, args.autopilot)

        clock = pygame.time.Clock()
        with open('log.csv', 'w', newline='') as csvfile:
            csvwriter = csv.writer(csvfile)
            csvwriter.writerow(['time'])
            while True:
                clock.tick_busy_loop(60)
                new_time = time.time()
                if controller.parse_events(client, world, clock):
                    return
                world.tick(clock)
                world.render(display)
                pygame.display.flip()
                csvwriter.writerow([str(new_time)]) 
```

With this we expect the interval between the time instants logged onto 'log.csv' to match with the Frame Rate we have set. But that is not what we see when we inspect the csv file. 

Could you please provide us with more clarity on these aspects.
